File: Gemini_Flash/Java/Factory_Method/round3/test/test_cafe3.java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class CafeOrderTest {

    @Test
    void testCreateItem_Beverage_Coffee() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Beverage", "Espresso");
        assertTrue(item instanceof Coffee);
        assertEquals("Espresso", ((Coffee) item).type);
    }

    @Test
    void testCreateItem_Beverage_Tea() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Beverage", "Green Tea");
        assertTrue(item instanceof Tea);
        assertEquals("Green Tea", ((Tea) item).type);
    }

    @Test
    void testCreateItem_Beverage_Cold() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Beverage", "Fruit Smoothies");
        assertTrue(item instanceof ColdBeverage);
        assertEquals("Fruit Smoothies", ((ColdBeverage) item).type);
    }

    @Test
    void testCreateItem_Pastry_Bread() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Pastry", "Croissant");
        assertTrue(item instanceof Bread);
        assertEquals("Croissant", ((Bread) item).type);
    }

    @Test
    void testCreateItem_Pastry_Cake() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Pastry", "Chocolate Cake");
        assertTrue(item instanceof Cake);
        assertEquals("Chocolate Cake", ((Cake) item).type);
    }

    @Test
    void testCreateItem_Pastry_Cookies() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Pastry", "Chocolate Chip Cookies");
        assertTrue(item instanceof Cookies);
        assertEquals("Chocolate Chip Cookies", ((Cookies) item).type);
    }

    @Test
    void testCreateItem_Snack_Sandwiches() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Snack", "Chicken Sandwich");
        assertTrue(item instanceof Sandwiches);
        assertEquals("Chicken Sandwich", ((Sandwiches) item).type);
    }

    @Test
    void testCreateItem_Snack_Pasta() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Snack", "Spaghetti");
        assertTrue(item instanceof Pasta);
        assertEquals("Spaghetti", ((Pasta) item).type);
    }

    @Test
    void testCreateItem_Snack_Salads() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Snack", "Vegetable Salad");
        assertTrue(item instanceof Salads);
        assertEquals("Vegetable Salad", ((Salads) item).type);
    }

    @Test
    void testCreateItem_Invalid() {
        CafeOrder order = new CafeOrder();
        MenuItem item = order.createItem("Invalid", "Invalid");
        assertNull(item);
    }

    @Test
    void testAddItem() {
        CafeOrder order = new CafeOrder();
        order.addItem("Beverage", "Espresso");
        assertEquals(1, order.order.size());
        assertTrue(order.order.get(0) instanceof Coffee);
        assertEquals("Espresso", ((Coffee) order.order.get(0)).type);
    }

    @Test
    void testPrepareOrder() {
        CafeOrder order = new CafeOrder();
        order.addItem("Beverage", "Espresso");
        order.addItem("Pastry", "Chocolate Cake");
        order.addItem("Snack", "Chicken Sandwich");
        order.prepareOrder();

        String expectedOutput = "กำลังเตรียม Espresso กาแฟ\n" +
                "กำลังเตรียม Chocolate Cake เค้ก\n" +
                "กำลังเตรียม Chicken Sandwich แซนวิช\n";
        assertEquals(expectedOutput, getOutputFromSystemOut());
    }

    // วิธีการช่วยในการรับข้อความจาก System.out
    private String getOutputFromSystemOut() {
        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
        System.setOut(new PrintStream(outputStream));
        return outputStream.toString();
    }
}

File: Gemini_Flash/Python/Factory_Method/round3/test/test_cafe3.py
import pytest
from your_module import (  # เปลี่ยน 'your_module' เป็นชื่อโมดูลของคุณ
    MenuItem, Espresso, Latte, Milkshakes, FruitJuices,
    Croissant, ChocolateCake, ChocolateChipCookies,
    TunaSandwich, Spaghetti, VegetableSalad,
    CafeOrder
)

def test_create_item_beverage():
    order = CafeOrder()
    espresso = order.create_item("Espresso")
    assert isinstance(espresso, Espresso)

def test_create_item_pastry():
    order = CafeOrder()
    chocolate_cake = order.create_item("Chocolate Cake")
    assert isinstance(chocolate_cake, ChocolateCake)

def test_create_item_snack():
    order = CafeOrder()
    tuna_sandwich = order.create_item("Tuna Sandwich")
    assert isinstance(tuna_sandwich, TunaSandwich)

def test_create_item_invalid():
    order = CafeOrder()
    invalid_item = order.create_item("Invalid")
    assert invalid_item is None

def test_add_item():
    order = CafeOrder()
    order.add_item("Espresso")
    assert isinstance(order.order[0], Espresso)

def test_serve_order():
    order = CafeOrder()
    order.add_item("Espresso")
    order.add_item("Chocolate Cake")
    order.add_item("Tuna Sandwich")
    order.serve_order()
    assert "เสิร์ฟ เอสเปรสโซ" in capsys.readouterr().out
    assert "เสิร์ฟ เค้กช็อกโกแลต" in capsys.readouterr().out
    assert "เสิร์ฟ แซนวิชทูน่า" in capsys.readouterr().out